### DISCUSSION TITLE
List Of Games That Use Channel/Texture Shuffle

### DISCUSSION THREAD

**Discussion by tadanokojin** [PCSX2/pcsx2#4109] [Information]
I want a list of games with shuffle post-processing including dumps.
Goal is to know which stuff to test if we change the handling.

[test plugin](https://github.com/PCSX2/pcsx2/files/5782402/GSdx32-SSE4-shuffle-log.zip)

## Definitions
**Texture Shuffle:** A channel copy that involves reading a 32-bit color/depth buffer as 16-bit and texture mapping 8xh pixel strips to the right or left by 8, skipping 8 and then repeating. The result is red+green can be mapped into blue+alpha or vice versa.
**Channel Shuffle:** A channel copy that involves reading a 32-bit color/depth buffer as 8-bit index texture and using a palette to modulate the color channels individually. The result is any individual channel can be copied or modified. [More Info](https://pcsx2.net/developer-blog/277-channel-shuffle-effect.html)
**Pixel Shuffle:** A special variant of texture shuffle that involves channel copying the red-green and blue-alpha pixels into a temporary buffer. [More Info](https://gist.github.com/tadanokojin/71ec3c3781a185befb5d552e0c62e555)

## Texture Shuffle
| Game | Dumps | Goal | Notes |
|-|-|-|-|
| [007 From Russia with Love](https://wiki.pcsx2.net/From_Russia_With_Love) | | | https://github.com/PCSX2/pcsx2/issues/3875 |
| [Big Mutha Truckers](https://wiki.pcsx2.net/Big_Mutha_Truckers) | [link 1](https://dumps.pcsx2.net/index.php/s/nNCTwbboyaqjNCy) | | https://github.com/PCSX2/pcsx2/issues/2449 |
| [Burnout 3: Takedown](https://wiki.pcsx2.net/Burnout_3:_Takedown) | | | |
| [Burnout: Revenge](https://wiki.pcsx2.net/Burnout_Revenge) | | | depth |
| [Crash Twinsanity](https://wiki.pcsx2.net/Crash_Twinsanity) | [link 1](https://dumps.pcsx2.net/index.php/s/TJfZoonJHoKd8gF) | Fog | half-screen, depth |
| [Devil May Cry 3](https://wiki.pcsx2.net/Devil_May_Cry_3) | | | |
| [Dog's Life](https://wiki.pcsx2.net/Dog%27s_Life) | | Depth of field | depth |
| [Dragonball Z: Budokai Tenkaichi 3](https://wiki.pcsx2.net/Dragon_Ball_Z:_Budokai_Tenkaichi_3) | | | |
| [Godfather](https://wiki.pcsx2.net/The_Godfather) | [link 1](https://dumps.pcsx2.net/index.php/s/3KnKKYkGb7qMjDy) | Fog | half-screen, depth, https://github.com/PCSX2/pcsx2/issues/3875 |
| [God of War](https://wiki.pcsx2.net/God_of_War) | | | | | |
| [Haunting Ground](https://wiki.pcsx2.net/Haunting_Ground) | [link 1](https://dumps.pcsx2.net/index.php/s/7wTgkBB3PrqsGgg) | Fog | depth |
| [Hitman: Contracts](https://wiki.pcsx2.net/Hitman:_Contracts) | | | |
| [Need for Speed: Most Wanted](https://wiki.pcsx2.net/Need_for_Speed_Most_Wanted) | | | |
| [Need for Speed: Hot Pursuit 2](https://wiki.pcsx2.net/Need_for_Speed:_Hot_Pursuit_2) | | | https://github.com/PCSX2/pcsx2/issues/2942 |
| [Resident Evil 4](https://wiki.pcsx2.net/Resident_Evil_4) | | | in-place |
| [Sly 2](https://wiki.pcsx2.net/Sly_2:_Band_of_Thieves) | | | |
| [Sly 3](https://wiki.pcsx2.net/Sly_3:_Honor_Among_Thieves) | | | |
| [Sonic Unleashed](https://wiki.pcsx2.net/Sonic_Unleashed) | [link 1](https://dumps.pcsx2.net/index.php/s/W7b5pcJfg8KHpy9) | | https://github.com/PCSX2/pcsx2/pull/2998 |
| [Speed Kings](https://wiki.pcsx2.net/Speed_Kings) | | Fog | half-screen, depth |
| [Transformers](https://wiki.pcsx2.net/Transformers) | | | https://github.com/PCSX2/pcsx2/issues/3025 |
| [Vampire Panic](https://wiki.pcsx2.net/Vampire_Panic) | | | sprite deswizzle, https://github.com/PCSX2/pcsx2/issues/3610 |

## Channel Shuffle
| Game | Dumps | Goal | Notes |
|-|-|-|-|
| [Global Defense Force](https://wiki.pcsx2.net/Simple_2000_Series_Vol._81:_The_Chikyuu_Boueigun_2) | | | in-place|
| Gran Turismo | [link 1](https://dumps.pcsx2.net/index.php/s/63ZpgJBTrZHjtH2) | Gamma Correction | |
| [Hitman: Blood Money](https://wiki.pcsx2.net/Hitman:_Blood_Money) | | Dot 3 | |
| Metal Gear Solid 2 | | | |
| Metal Gear Solid 3 | | | |
| Raw Danger (Demo A) | | | SCANMSK, depth, https://github.com/PCSX2/pcsx2/issues/2753|
| Ridge Racer V | | Gamma Correction | |
| Tales of Abyss | | | depth, 16-bit |
| Terminator 3 | | | | |
| Urban Chaos | [link 1](https://dumps.pcsx2.net/index.php/s/GnBzdEiFjr63zSo) | Fog | depth, 16-bit |

## Combination
| Game | Dumps | Goal | Notes |
|-|-|-|-|
| Destroy All Humans | | | |
| Ghost In The Shell SAC | | | |
| PachiPara 14 | | | pixel shuffle |
| Raw Danger | [link 1](https://dumps.pcsx2.net/index.php/s/CP4TXsdM9wqnsqg) | Fog, Gamma Correction | pixel shuffle |
| Steambot Chronicles | [link 1](https://dumps.pcsx2.net/index.php/s/gDp9nDDP6gR83m2) | | pixel shuffle |

## TODO
- valkyrie profile 2 (I think just channel but don't remember)

**Comment by refractionpcsx2** [comment]
[BiA-RH30-emuLog.zip](https://github.com/PCSX2/pcsx2/files/5782575/BiA-RH30-emuLog.zip)

Brothers in Arms - Road to Hill 30

[Burnout3- emuLog.zip](https://github.com/PCSX2/pcsx2/files/5782506/Burnout3-.emuLog.zip)

Burnout 3

[DestroyAllHumans-emuLog.zip](https://github.com/PCSX2/pcsx2/files/5782469/DestroyAllHumans-emuLog.zip)

Destroy All Humans log, noting this only seems to throw log messages out during the intro, ingame does not (possibly missed effect?)

[GodOfWar-emuLog.zip](https://github.com/PCSX2/pcsx2/files/5782524/GodOfWar-emuLog.zip)

God Of War

[MGS3-Start-emuLog.zip](https://github.com/PCSX2/pcsx2/files/5783721/MGS3-Start-emuLog.zip)

Metal Gear Solid (Start of game, no thermal goggles)

[NFS-MostWanted-emuLog.zip](https://github.com/PCSX2/pcsx2/files/5782560/NFS-MostWanted-emuLog.zip)

Need For Speed Most Wanted

[ValkyrieProfile2-emuLog.zip](https://github.com/PCSX2/pcsx2/files/5782546/ValkyrieProfile2-emuLog.zip)

Valkyrie Profile 2 (Only during the post menu rendered cutscene, ingame spams nothing)

**Comment by Dogmander** [comment]
Dog's Life dump
[DogsLife-emuLog.zip](https://github.com/PCSX2/pcsx2/files/6104519/DogsLife-emuLog.zip)
